app/src/main/java/com/example/background/BlurApplication.kt

```diff
@@ -17,14 +17,29 @@
 package com.example.background
 
 import android.app.Application
-import timber.log.Timber
-import timber.log.Timber.DebugTree
+import androidx.work.Configuration
 
-class BlurApplication() : Application() {
+/**
+ * This is the Application class that's instantiated before any other class when the application is
+ * first created. Here we are using it to set configurations for WorkManager in order to see more
+ * verbose logging related to our work requests. Read more about custom WorkManager configuration
+ * here: https://developer.android.com/topic/libraries/architecture/workmanager/advanced/custom-configuration
+ */
+class BlurApplication() : Application(), Configuration.Provider {
     override fun onCreate() {
         super.onCreate()
-        if (BuildConfig.DEBUG) {
-            Timber.plant(DebugTree())
+    }
+
+    override fun getWorkManagerConfiguration(): Configuration {
```

Comment:
For Android Basics students, [On-Demand Initialization](https://developer.android.com/topic/libraries/architecture/workmanager/advanced/custom-configuration#on-demand) is an advanced concept. IMHO It is unlikely that they will understand.

Reply:
Yeah, I agree, it's quite advanced. My thought for keeping it in was that this class will already be provided in full as part of the starter code, so students won't have to work with this class at all. The benefit is that if they go into Logcat, they'll be able to see more verbose logs related to their work requests. Do you think that would be okay? Or should we just take out this class altogether?

Reply:
I see, if you think it is okay to skip this, that would be great. My preference would be to remove the code altogether. Alternatively, we can give this code in an optional section in the codelab but not in the repo.  
If you feel strongly that this needs to be included, then let's add some comments and point them to the documentation. LMKWYT.

Reply:
Okay! We'll take this class out altogether then. The [advanced codelab](https://developer.android.com/codelabs/android-adv-workmanager) goes through custom configurations in detail, so we can always point to the advanced codelab at the end of this one. Thanks Jhansi!